mutiny-core/src/nostr/nwc.rs

```diff
@@ -210,15 +237,19 @@ impl NostrWalletConnect {
             // if we need approval, just save in the db for later
             match self.profile.spending_conditions.clone() {
                 SpendingConditions::SingleUse(mut single_use) => {
-                    // check if we have already spent
-                    if single_use.spent {
-                        return Ok((None, needs_save));
-                    }
-
                     let msats = invoice.amount_milli_satoshis().unwrap();
 
-                    // verify amount is under our limit
-                    let content = if msats <= single_use.amount_sats * 1_000 {
+                    // check if we have already spent
+                    let content = if single_use.spent {
+                        Response {
+                            result_type: Method::PayInvoice,
+                            error: Some(NIP47Error {
+                                code: ErrorCode::QuotaExceeded,
+                                message: "Already Claimed".to_string(),
+                            }),
+                            result: None,
+                        }
+                    } else if msats <= single_use.amount_sats * 1_000 {
```

Comment:
I think `todo handle timeout errors` need to be handled. I think there's restart risk with in flight payments being redeemed and the NWC profile isn't saved because that happens only here.

Like maybe check inflight payments somehow and see if there's any associated with this specific NWC profile?

Reply:
We currently check if we've paying an invoice already. We only allow retrying if the payment status is failed

https://github.com/MutinyWallet/mutiny-node/blob/5c8288695d520b4e8252c86e2580cba4eb1aea2d/mutiny-core/src/node.rs#L945

Reply:
It's not about paying the same invoice, it's about another NWC message coming in with a new invoice. If the payment stalls out, or it completes outside of this process running, then the NWC profile is never marked as paid. Another NWC message can come in with a new invoice and repeat the process.

Reply:
Yeah this is defnitely an issue but not sure the best way to go about it without having to tie this into the node's event manager.

For now just having it log and disable the nwc profile on payment timeout to prevent double spends.

Reply:
Okay, that's a good approach for now. I think long term we can save pending NWC invoices and have those get checked before attempting to pay another for the same NWC. We will need that for auto approval w/ budget too.